publish comparison between LMDB and LevelDB

mykmelez opened the pull request:
@joewalker This is a Markdown version of the Google Doc I wrote a few months ago to compare LMDB and LevelDB. I tried to format it in the same two-column format I used in the Google Doc, with a side-by-side comparison of the two storage engines within each section; but GitHub-flavored Markdown makes that hard, so I eventually gave up and restructured the doc to describe first LMDB and then LevelDB within each section. Is there anything else that you recommend we change about the content or formatting of this doc?

@ncalexan You previously looked at the Google Doc and had some feedback, after which I made some changes. I've since added another footnote about Wikipedia reliability and the use of it as one (but not the only) source of information about the two storage engines. Is there anything you would change before we publish this?

ncalexan commented on text/0017-lmdb-vs-leveldb.md:
```diff
@@ -0,0 +1,238 @@
+
+#### LevelDB
+
+LevelDB was "written at Google" (according to its <a href="https://github.com/google/leveldb">GitHub project page</a>) and "inspired by <a href="https://en.wikipedia.org/wiki/Bigtable">BigTable</a>" (according to its <a href="https://en.wikipedia.org/wiki/LevelDB">Wikipedia entry</a><sup>[1](#footnote-1)</sup>), although it doesn't share code with that proprietary storage system. It was originall ...
```
nit: the capitalization of "bigtable" differs in the link and the anchor.  Intentional?

mykmelez replied:
Unintentional! It looks like the project capitalizes its name "Bigtable," so I've fixed the capitalization in 811db34.

ncalexan commented on text/0017-lmdb-vs-leveldb.md:
I'm going to guess it wasn't released in 2038 :)

ncalexan commented on text/0017-lmdb-vs-leveldb.md:
Hmm, maybe a find and replace gone wrong?  2034 and 2037 are _very_ future oriented :)  🤖

ncalexan commented on text/0017-lmdb-vs-leveldb.md:
nit: back quotes around code snippets?

mykmelez replied:
Yes, good idea, added in d8b2354.

ncalexan commented on text/0017-lmdb-vs-leveldb.md:
nit: LevelDB (capitalization)?

mykmelez replied:
This is actually how the blog post in question capitalizes LevelDB in its title.

(In the body of the blog post, the author uses all-lower-case "leveldb" with its first letter capitalized only at the beginning of a sentence, à la "sentence case" with the word treated as an improper noun.)

ncalexan commented on text/0017-lmdb-vs-leveldb.md:
Stripping symbols will make more difference than optimizations.  Stripped?  Maybe include the Rust `[profile]` block, or link to it?  These comparisons are inherently difficult to reproduce, I know :)

mykmelez replied:
I rebuilt and stripped the executables, which changed the numbers a bit but didn't alter the relative differences much. I don't have a [profile] block, but I added to the footnotes the Rust version and a note that I used the "release" profile and stripped the executables.

mykmelez commented:
> Looks good! There's a systemic issue with dates -- I'd look for everything with "203" or "20" in it -- but otherwise I like it!

It looks like I accidentally did `s/1/3/` across the entire document, apparently when I was linkifying footnotes, so it isn't only dates.

In any case, I fixed the dates in 0eeb358, then I found the errant commit in the branch history (from before I squashed the branch in preparation to request a pull) and reverted it, fixing the non-date errors in da9a18d.
